lib/tasks/services.rake

```diff
@@ -12,4 +12,15 @@ namespace :services do
   task :destroy_marked_as_deleted => :environment do
     DestroyAllDeletedObjectsWorker.perform_later(Service.to_s)
   end
+
+  desc 'Destroy a default service.'
+  task :destroy_default_service, %i[account_id service_id] => [:environment] do |_task, args|
+    account = Account.providers.find(args[:account_id])
+    service_id = args[:service_id]
+    service = account.services.find(service_id) # So it will raise an error if this service does not belong to this account :)
+    new_default_service = account.services.where.not(id: service_id).select(:id).first! # It will raise an error if it does not have another service
+    account.update!({default_service_id: nil}, without_protection: true) # We removed this flag and should never add it anymore ;) we only keep it for the old ones who already have it
```

Comment:
The opinion of each and every person of @3scale/system-ruby is pure gold for this 😉

Reply:
what about this compromise:

```
default_service_id = account.default_service_id.present? ? new_default_service.id  : nil
account.update!({default_service_id: default_service_id}, without_protection: true)
```

so we wouldn't add default_service_id for new accounts and keep it for the old ones.

Reply:
mmmm the goal of this task is to be called only if it is a default service :thinking: but you are right that it apparently can be called if it is not default, but then, as it is not what it is for, it would mess up badly even in Apisonator XD so... oki, I will change it to work for both default and non-default services :wink: